test/e2e/spec/shared_spec.rb

```diff
@@ -249,9 +248,246 @@
+                                          { "cosigner#0" => "self" },
+                                        "template" =>
+                                            { "all" =>
+                                               [ "cosigner#0",
+                                                 { "active_from" => 120 }
+                                               ]
+                                            }
+                                    }
+
+        delegation_script_template = { 'cosigners' =>
+                                            { 'cosigner#0' => "self" },
+                                        'template' =>
+                                            { 'all' =>
+                                               [ 'cosigner#0',
+                                                 'cosigner#1'
+                                               ]
+                                            }
+                                      }
+        payload = { account_public_key: acc_xpub,
+                    passphrase: PASS,
+                    name: "Shared wallet",
+                    account_index: acc_ix,
+                    payment_script_template: payment_script_template,
+                    delegation_script_template: delegation_script_template,
+                    }
+
+        wallet = w.create(payload)
+        expect(wallet).to be_correct_and_respond 201
+
+        wid = wallet['id']
+        g = w.get(wid)
+        expect(g).to be_correct_and_respond 200
+
+        expect(w.delete(wid)).to be_correct_and_respond 204
+      end
     end
 
+    describe "Public Keys" do
+      matrix = {
+        "utxo_internal" => "addr_shared_vk",
+        "utxo_external" => "addr_shared_vk",
+        "mutable_account" => "stake_shared_vk"
+      }
+      matrix_h = {
+        "utxo_internal" => "addr_shared_vkh",
+        "utxo_external" => "addr_shared_vkh",
+        "mutable_account" => "stake_shared_vkh"
+      }
+
+      it "Get public key - pending wallet from mnemonics" do
+        pending "utxo_internal returns empty response"
+        m24 = mnemonic_sentence(24)
+        acc_xpub = cardano_address_get_acc_xpub(m24, "1854H/1815H/0H")
+        pending_wid = create_pending_shared_wallet(m24, '0H', acc_xpub)
 
-  end
+        matrix.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(pending_wid, role, id)
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+
+        matrix_h.each do |role, addr_prefix|
+          id = [*0..100000].sample
+          res = SHARED.keys.get_public_key(pending_wid, role, id, {hash: true})
+          expect(res).to be_correct_and_respond 200
+          expect(res.to_s).to include addr_prefix
+        end
+      end
+
+      it "Get public key - pending wallet from acc pub key" do
+        pending "utxo_internal returns empty response"
```

Comment:
@paweljakubas I've added tests covering the issue where utxo_internal returns empty response and tagged them pending for now...